server/test/controllers/admin/AdminProgramImageControllerTest.java

```diff
@@ -369,33 +370,111 @@ public void updateFileKey_incorrectlyFormatted_throwsException() {
         .withMessageContaining("Key incorrectly formatted");
   }
 
+  @Test
+  public void updateFileKey_hasBucketAndKey_keyUpdated() throws ProgramNotFoundException {
```

Comment:
question (non-blocking): is this test related to these changes or did you realize you should have had it from previous changes? is this test testing updating a key or adding one or is there no difference?

Reply:
This test isn't related to this PR -- I just realized that I should've wrote it previously but didn't.

It's testing adding a key, not updating one (the controller method is just called `updateFileKey`). I added a test for updating a key. 

Also let me know if I should move adding these tests to a separate PR! Happy to do so.

Reply:
I think this is fine! Thanks for explaining :)